src/Builder/Query.php

```diff
@@ -40,6 +44,21 @@ public static function regex(Regex|string $regex, string|null $flags = null): Re
         return self::generatedRegex($regex);
     }
 
+    /**
+     * The $elemMatch operator matches documents that contain an array field with at least one element that matches all the specified query criteria.
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/
+     */
+    public static function elemMatch(QueryInterface|FieldQueryInterface|Type|stdClass|array|bool|float|int|string|null ...$query): ElemMatchOperator
+    {
+        // $elemMatch accept a query or a field query
+        if (array_is_list($query)) {
+            return self::generatedElemMatch(new CombinedFieldQuery($query));
+        }
```

Comment:
What syntax is this referring to? It's not clear to me from looking at [$elemMatch: Syntax](https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/#syntax) and the [Single Query Condition](https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/#std-label-elemmatch-single-query-condition) section linked therein. IIRC, the value for an `$elemMatch` is always just a query object.

Reply:
The `CombinedFieldQuery` is to support the [Element Match](https://www.mongodb.com/docs/manual/reference/operator/query/elemMatch/#element-match) example. There is no field name in the `$elemMatch` value :
```ts
{
  results: {
    $elemMatch: {
      $gte: 80,
      $lt: 85
  }
}
```

Without this code, the generated pipeline look like this:
```ts
{
  results: {
    $elemMatch: {
      "0": { $gte: 80 },
      "1": { $lt: 85 }
    }
  }
}
```

I added a condition so that we accept a `QueryInterface` object if someone needs to use `0` as field name.

Reply:
Ah, for matching an element in an array of non-document values. That makes sense.

I think that'd be helpful to mention in the comment here (or maybe refer to "element match").